refactor: TestWallet API cleanup pt.2

benesjan opened the pull request:
Continuation of https://github.com/AztecProtocol/aztec-packages/pull/17283. Changes explained with comments in the code.

benesjan commented:
* **#17380** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17380?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17325** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17325?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17295** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17295?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17295?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#17303** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17303?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17283** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17283?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

benesjan commented on yarn-project/aztec.js/src/wallet/account_manager.ts:
```diff
@@ -103,11 +103,16 @@ export class AccountManager {
     return this.instance.address;
   }
 
+  get address() {
+    return this.instance.address;
+  }
```
As discussed in DMs, this getter just looks nicer in code.

Thunkar replied:
I'll probably kill the one above soon, since we're "starting from scratch" on wallet apis

benesjan commented on docs/docs/the_aztec_network/reference/cli_reference.md:
```diff
@@ -827,9 +827,6 @@ If two subsystems can contain the same configuration option, only one needs to b
     --bot.senderSalt <value>                                                                                                      ($BOT_ACCOUNT_SALT)
           The salt to use to deploys the sender account.
 
-    --bot.recipientEncryptionSecret <value>                        (default: [Redacted])                                         ($BOT_RECIPIENT_ENCRYPTION_SECRET)
```
Alex confirmed that dropping this and just using arbitrary recipient is fine.

benesjan commented on docs/docs/developers/docs/concepts/call_types.md:
```diff
@@ -189,7 +189,7 @@ No correctness is guaranteed on the result of `simulate`! Correct execution is e
 This creates and returns a transaction request, which includes proof of correct private execution and side-effects. The request is not broadcast however, and no gas is spent. It is typically used in testing contexts to inspect transaction parameters or to check for execution failure.
 
-#include_code local-tx-fails /yarn-project/end-to-end/src/guides/dapp_testing.test.ts typescript
+#include_code local-tx-fails /yarn-project/end-to-end/src/e2e_crowdfunding_and_claim.test.ts typescript
```
Dropped the `dapp_testing.test.ts ` because it was stale and basically no longer used in docs. This was the only occurrence in docs.

What motivated me to drop it is that we directly obtained there the notes from wallet and that is deprecated.

benesjan commented on yarn-project/end-to-end/src/e2e_crowdfunding_and_claim.test.ts:
```diff
@@ -95,7 +95,10 @@ describe('e2e_crowdfunding_and_claim', () => {
     );
     const crowdfundingInstance = await crowdfundingDeployment.getInstance();
-    await wallet.registerAccount(crowdfundingSecretKey, await computePartialAddress(crowdfundingInstance));
+    await wallet.registerKeysForEscrowContract(
+      crowdfundingSecretKey,
+      await computePartialAddress(crowdfundingInstance),
+    );
```
Renamed the problematic function to make it less cancerous before it gets completely nuked.

benesjan commented on yarn-project/end-to-end/src/e2e_bot.test.ts:
```diff
@@ -67,12 +67,11 @@ describe('e2e_bot', () => {
 
-    it('reuses the same account and token contract', async () => {
-      const { defaultAccountAddress, token, recipient } = bot;
+    it('reuses the same token contract', async () => {
```
Recipient is no longer stable across bot instantiations but Alex said it doesn't matter.

Thunkar commented on yarn-project/test-wallet/src/wallet/test_wallet.ts:
```diff
@@ -217,9 +234,11 @@ export abstract class BaseTestWallet extends BaseWallet {
+   * Returns the PXE.
+   * @deprecated This is only used by account manager to because there we call registerAccount. This can be dropped
+   * once we allow Wallet.registerContract accepts secretKey and partialAddress on the input.
+   */
```
I'm killing this next, so we should be good

benesjan committed:
refactor: TestWallet API cleanup pt.2

Continuation of https://github.com/AztecProtocol/aztec-packages/pull/17283. Changes explained with comments in the code.
